engine/src/main/java/org/terasology/particles/ParticleSystemManagerImpl.java

```diff
@@ -107,11 +111,16 @@ public void update(float delta) {
     @Override
     public Stream<ParticleRenderingData> getParticleEmittersByDataComponent(Class<? extends Component> particleDataComponent) {
         return particleUpdater.getParticleEmitters().stream()
-                .filter(p -> p.ownerEntity.hasComponent(particleDataComponent))
-                .map(particleEmitterComponent ->
-                        new ParticleRenderingData<>(
-                                particleEmitterComponent.ownerEntity.getComponent(particleDataComponent),
-                                particleEmitterComponent.particlePool
-                        ));
+                .filter(emitter -> emitter.ownerEntity.hasComponent(particleDataComponent))
+                .filter(distinctByKey(emitter -> emitter.particlePool))
```

Comment:
Here are some comments that should clarify the functionality of each filter. 

```suggestion
                .filter(emitter -> emitter.ownerEntity.hasComponent(particleDataComponent)) // check if owning entity has particleDataComponent
                .filter(distinctByKey(emitter -> emitter.particlePool)) // each particle emitter has a unique pool
```